install/rctf.py

```diff
@@ -0,0 +1,163 @@
+        
+        if not verify_privileges():
+            return False
+
+        if not execute('docker-compose down'):
+            logging.fatal('Failed to stop rCTF instance')
+            return False
+        
+        return True
+    
+    
+    def upgrade(self):
+        os.chdir(self.install_path)
+        
+        if not verify_privileges():
+            return False
+    
+        # XXX: is there a way to make this not error if it fails?
+        self.down()
+        
+        if not execute('git checkout %s' % REPOSITORY_BRANCH):
+            logging.fatal('Failed to checkout repository branch %s' % REPOSITORY_BRANCH)
+            return False
+
+        if not execute('git pull'):
+            logging.fatal('Failed to pull latest from repository')
+            return False
+
+        if not execute('docker-compose build --no-cache'):
+            logging.fatal('Failed to rebuild docker image')
+            return False
+        
+        return True
+
+
+
+# main
+
+
+if __name__ == '__main__':
+    # parse arguments
+
+
+    parser = argparse.ArgumentParser(description = 'Manage rCTF installations from the CLI')
+    
+    parser.add_argument('--install-path', type = str, default = os.environ.get('RCTF_INSTALL_PATH', os.environ.get('INSTALL_PATH', '/opt/rctf/')), help = 'The path to the rCTF installation to manage')
+
+    subparsers = parser.add_subparsers(help = 'The sub-command to execute')
+
+    parser_up = subparsers.add_parser('up', help = 'Start rCTF in background')
+    parser_up.set_defaults(subcommand = 'up')
+    
+    parser_down = subparsers.add_parser('down', help = 'Stop rCTF if running')
+    parser_down.set_defaults(subcommand = 'down')
+
+    parser_update = subparsers.add_parser('update', help = 'Update the rCTF installation')
+    parser_update.set_defaults(subcommand = 'update')
+
+    args = parser.parse_args()
+
+    
+    if not 'subcommand' in vars(args):
+        logging.info('This is an rCTF management script. For usage, run:\n    %s --help' % sys.argv[0])
+        exit(0)
+    
+
+    # create instance
+
+
+    install_path = args.install_path
+    subcommand = args.subcommand
+
+
+    rctf = rCTF(install_path = install_path)
+
+    os.chdir(install_path)
+
+
+    if subcommand in ['start', 'up']:
+        if not rctf.up():
+            exit(1)
+    elif subcommand in ['stop', 'down']:
+        if not rctf.down():
+            exit(1)
+    elif subcommand in ['update', 'upgrade']:
+        if not rctf.upgrade():
+            exit(1)
+
+        logging.info('Successfully updated instance. Run `%s up` to start instance' % sys.argv[0])
+        logging.info('Upgrading %s CLI tool...' % sys.argv[0])
+        
+        # XXX: command injection? i can't think of a practical scenario here
+        # TODO/BUG: either way there is the possibility an admin could have a directory with a name containing a "
+        if not execute('cp install/rctf.py "%s" && chmod + "%s"' % (sys.argv[0], sys.argv[0])):
```

Comment:
We might as well patch it, shouldn't be too hard to escape shell quotes? see: https://stackoverflow.com/questions/35817/how-to-escape-os-system-calls

Reply:
Yup, I marked this as a draft because there's a lot more to do before merging. I just didn't want to get caught up in details yet cause there's still a lot to write for this PR.